assertions/manager.go

```diff
@@ -256,49 +307,13 @@ func (m *Manager) AssertionsSubmittedInProcess() []common.Hash {
 	return hashes
 }
 
-func (m *Manager) logChallengeConfigs(ctx context.Context) error {
-	cm, err := m.chain.SpecChallengeManager(ctx)
-	if err != nil {
-		return err
-	}
-	bigStepNum, err := cm.NumBigSteps(ctx)
-	if err != nil {
-		return err
-	}
-	challengePeriodBlocks, err := cm.ChallengePeriodBlocks(ctx)
-	if err != nil {
-		return err
-	}
-	layerZeroHeights, err := m.LayerZeroHeights(ctx)
-	if err != nil {
-		return err
-	}
-	log.Info("Opening challenge with the following configuration",
-		"address", cm.Address(),
-		"bigStepNumber", bigStepNum,
-		"challengePeriodBlocks", challengePeriodBlocks,
-		"layerZeroHeights", layerZeroHeights,
-	)
-	return nil
-}
-
 // Returns true if the manager can respond to an assertion with a challenge.
 func (m *Manager) canPostRivalAssertion() bool {
```

Comment:
Hmmm. I can sort of make arguments in either direction. I think this is a very subjective thing.

On the one hand, `canPostRivalAssertion` is only an interesting question in the context of an abstraction which is responsible for posting rival assertions (if it's allowed.) So, while it needs to know which "Mode" it's in, the assertion manager seems like the "right" abstraction for answering this question.

By the same logic, the `canPostChallenge` really belongs in the ChallengeManager abstraction. And, in fact, I *think* it could be moved there because it is only called to stop the AssertionManager from calling `rivalHandler.HandleCorrectRival` (and a few other lines of pre-challenge checks.) So, we could just unconditionally call `HandleCorrectRival` and let the ChallengeManager check the mode to decide if it should actually post a new challenge edge or not.

Arguing for moving these methods onto the Mode type, you could definitely say, well, the results of the calls are a pure function of the mode.  It might be a little more obvious to readers of the code if the calls looked like this: `if mode.SupportsRivalAssertions()` or `if !mode.SupportsChallengeEdges()`. Those make it obvious that if the system isn't doing a thing the programmer thought it should be doing, it's likely because it's configured in the wrong mode (without having to read the implementation of a method like "canPostRivalAssertions".)

For now, I'll move the methods to the Mode type. But, what do you think of also moving the logic for checking if the mode supports opening challenge edges to the challenge manager and "always" calling the `HandleCorrectRival` function after creating a rival assertion?